src/classes/Bucket.ts

```diff
@@ -138,6 +156,8 @@ export class Bucket {
   ) {
     const contractPoolWithSigner = this.poolContract.connect(signer);
 
+    await this.validateLPBalance();
```

Comment:
That's not right; should be validating the `toIndex` bucket, not the current one.